atc/api/jobserver/create_build.go

```diff
@@ -34,7 +36,8 @@ func (s *Server) CreateJobBuild(pipeline db.Pipeline) http.Handler {
 			return
 		}
 
-		build, err := job.CreateBuild()
+		acc := accessor.GetAccessor(r)
+		build, err := job.CreateBuild(acc.Claims().UserID)
```

Comment:
Just checking: is `UserID` the correct value to use here for most connectors? I wonder how this value compares between the various connectors we have. I think I would expect e.g. my GitHub login, `vito`, to be stored when I trigger a build, but I don't know if that's `UserID`, `Username`, or `PreferredUsername`.

@xtremerui Does using `UserID` seem sensible to you?

Reply:
`UserID` could be a string that is not human readable so UserName or PreferredUserName should be used here. For example,

https://github.com/concourse/concourse/blob/d1dd60f7555f6e7f7d22b0e55289c24534c34b45/atc/api/accessor/accessor.go#L225-L231

Reply:
I tested with local user and LDAP:

* With local user, preferred_username is empty, `userid` and `username` are identical, in my test, both were `test`.
* With LDAP, preferred_username is also empty, `userid` is `chaol`, and `username` is `Chao (Evan) Li`. In LDAP case, `userid` is expected.

From https://github.com/concourse/concourse/blob/master/atc/api/accessor/accessor.go#L107-L118, both `userid` and `username` might be used. So I think the logic can be, if `userid` is not empty, then use `userid`, otherwise use `username`. @vito @xtremerui WDYT?

Reply:
Trying multiple makes sense to me - I'll leave it to @xtremerui to decide whether to include `preferred_username` as I don't know which connectors use it. :+1:

Devil's advocate: is there a standard convention for this kind of thing? Would it make sense to pass a structured value, since different resource types might need different fields? Should we just pass all the claims along? Is that crazy? (I don't hold any opinions here, I see pros and cons, but thought I'd bring it up in case anyone has a perspective to share.)

Reply:
Given we are not sure which id fits to all connectors and it's hard to test all connectors, passing entire claims sounds a good idea. In the first step, we can store entire claims in db, so that even if we only pass `userid` this time, it would be very easy to passing other fields without further db changes in future.